src/fenicsxconcrete/finite_element_problem/linear_elasticity_grf.py

```diff
@@ -0,0 +1,143 @@
+            self.L = self.L + external_force
+
+        body_force = self.experiment.create_body_force(self.v)
+        if body_force:
+            self.L = self.L + body_force
+
+        # boundary conditions only after function space
+        bcs = self.experiment.create_displacement_boundary(self.V)
+
+        self.a = ufl.inner(self.sigma(self.u_trial), self.epsilon(self.v)) * ufl.dx
+        self.weak_form_problem = df.fem.petsc.LinearProblem(
+            self.a,
+            self.L,
+            bcs=bcs,
+            petsc_options={"ksp_type": "preonly", "pc_type": "lu"},
+        )
+
+    # Random E and nu fields.
+    def random_field_generator(
+        self,
+        field_function_space,
+        cov_name,
+        mean,
+        correlation_length1,
+        correlation_length2,
+        variance,
+        no_eigen_values,
+        ktol,
+    ):
+        random_field = Randomfield(
+            field_function_space,
+            cov_name,
+            mean,
+            correlation_length1,
+            correlation_length2,
+            variance,
+            no_eigen_values,
+            ktol,
+        )
+        # random_field.solve_covariance_EVP()
+        return random_field
+
+    def parameters_conversion(self, lognormal_mean, lognormal_sigma):
+        from math import sqrt
+
+        normal_mean = np.log(lognormal_mean / sqrt(1 + (lognormal_sigma / lognormal_mean) ** 2))
+        normal_sigma = np.log(1 + (lognormal_sigma / lognormal_mean) ** 2)
+        return normal_mean, normal_sigma
+
+    def get_lames_constants(
+        self,
+    ):
+        # Random E and nu fields.
+        E_mean, E_variance = self.parameters_conversion(self.p["E"], 100e9)  # 3
+        Nu_mean, Nu_variance = self.parameters_conversion(self.p["nu"], 0.3)  # 0.03
+
+        self.E_randomfield = self.random_field_generator(
+            self.field_function_space, "squared_exp", E_mean, 0.3, 0.05, E_variance, 3, 0.01
+        )
+        self.E_randomfield.create_random_field(_type="random", _dist="LN")
+
+        self.nu_randomfield = self.random_field_generator(
+            self.field_function_space, "squared_exp", Nu_mean, 0.3, 0.05, Nu_variance, 3, 0.01
+        )
+        self.nu_randomfield.create_random_field(_type="random", _dist="LN")
+
+        lame1 = (self.E_randomfield.field.vector[:] * self.nu_randomfield.field.vector[:]) / (
+            (1 + self.nu_randomfield.field.vector[:]) * (1 - 2 * self.nu_randomfield.field.vector[:])
+        )
+        lame2 = self.E_randomfield.field.vector[:] / (2 * (1 + self.nu_randomfield.field.vector[:]))
+        return lame1, lame2
+
+    # TODO move this to sensor definition!?!?!
+    def pv_plot(self, t: int = 0) -> None:
+        # TODO add possibility for multiple time steps???
+        # Displacement Plot
+
+        # "Displacement.xdmf"
+        # pv_output_file
+        # TODO Look into how to unify in one file
+        with df.io.XDMFFile(self.mesh.comm, self.pv_output_file, "w") as xdmf:
```

Comment:
Potentially possible to use only one xdmf file

Reply:
During implementing the thixotropy problem, I came to the problem of saving time series in the pv_plot fct of the material_problem class.
I did same minor changes so that it works as before by doing:

- initializing the xdmf file in the init of the base material problem via
https://github.com/BAMresearch/FenicsXConcrete/blob/69-implementing-am-thixotropy-model/src/fenicsxconcrete/finite_element_problem/base_material.py

with df.io.XDMFFile(self.mesh.comm, self.pv_output_file, "w") as f:
            f.write_mesh(self.mesh)


- using append in the pv_plot fct of the actual material problem
https://github.com/BAMresearch/FenicsXConcrete/blob/69-implementing-am-thixotropy-model/src/fenicsxconcrete/finite_element_problem/concrete_am.py  

   with df.io.XDMFFile(self.mesh.comm, self.pv_output_file, "a") as f:
            f.write_function(self.displacement, _t)
            f.write_function(sigma_plot, _t)
            f.write_function(E_plot, _t)